File: java/src/test/java/problems101_200/ReorderListTest.java
package problems101_200;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ReorderListTest extends SampledTest {
    @Test
    public void test() {
        for (ReorderList solution : ReorderList.values()) {
            for (ReorderListSample sample : getSamples(ReorderListSample[].class)) {
                SinglyListNode input = SinglyListNode.of(sample.input.head);
                solution.reorderList(input);
                assertWithMessage(sample.getMessage())
                    .that(input)
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/ReorderListTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ReorderListTest : SampledTest() {
    @Test
    fun test() =
        ReorderList.entries.forEach { solution ->
            getSamples<Array<ReorderListSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(SinglyListNode.of(*input.head).also { solution.reorderList(it) })
                    .isEqualTo(SinglyListNode.of(*output))
            }
        }
}
